### PR TITLE
CLN: Some groupby internals

### PR DESCRIPTION
**Pull request by mroeschke**
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

1. Avoids calling `_get_sorted_data()` twice
2. Eliminates `*args, **kwargs` from an internal function

### CODE REVIEW

**Comment by WillAyd** [comment]
Looks like a segfault on travis - restarted to see

File: pandas/core/groupby/ops.py [hunk 1]
```diff
@@ -941,11 +939,13 @@ def _chop(self, sdata: DataFrame, slice_obj: slice) -> DataFrame:
             return sdata.iloc[:, slice_obj]
 
 
-def get_splitter(data: FrameOrSeries, *args, **kwargs) -> DataSplitter:
+def get_splitter(
+    data: FrameOrSeries, labels, ngroups: int, axis: int = 0
```
Comment (jbrockmendel) [inline pandas/core/groupby/ops.py]: is labels an Index?

File: pandas/core/groupby/ops.py [hunk 2]
```diff
@@ -927,11 +927,9 @@ def _chop(self, sdata: Series, slice_obj: slice) -> Series:
 
 
 class FrameSplitter(DataSplitter):
-    def fast_apply(self, f, names):
+    def fast_apply(self, f, sdata, names):
```
Comment (jbrockmendel) [inline pandas/core/groupby/ops.py]: is sdata a DataFrame?

**Comment by jbrockmendel** [comment]
potential annotations; LGTM pending green

**Comment by WillAyd** [comment]
Can you merge master? Slightly concerned about reproducibility of previous segfault

**Comment by mroeschke** [comment]
Thanks for the review. Merged master and all green

**Pull request merged**